packages/shared/src/util/features.ts

```diff
@@ -726,3 +745,97 @@ export function getParsedPrereqCondition(condition: string) {
   }
   return undefined;
 }
+function checkEnvironmentsMatch(
+  feature: FeatureInterface,
+  environments: string[],
+  reviewSetting: RequireReview
+) {
+  for (const env of reviewSetting.environments) {
+    if (environments.includes(env)) {
+      return true;
+    }
+  }
+  // check projects
+  if (feature?.project && reviewSetting.projects.includes(feature?.project)) {
+    return true;
+  }
+  for (const tag of reviewSetting.tags) {
+    if (feature?.tags?.includes(tag)) {
+      return true;
+    }
+  }
+  return (
+    reviewSetting.environments.length === 0 &&
+    reviewSetting.projects.length === 0 &&
+    reviewSetting.tags.length === 0
+  ); //if everything is empty is means it is on for all.
+}
```

Comment:
Add unit tests for all of these functions here.  The logic is tricky to follow and I'm noticing a lot of potential bugs